upup/pkg/fi/cloudup/bootstrapchannelbuilder/bootstrapchannelbuilder.go

```diff
@@ -848,8 +848,8 @@ func (b *BootstrapChannelBuilder) buildAddons(c *fi.CloudupModelBuilderContext)
 	}
 
 	// The metadata-proxy daemonset conceals node metadata endpoints in GCE.
-	// It will land on nodes labeled cloud.google.com/metadata-proxy-ready=true
-	if b.Cluster.Spec.GetCloudProvider() == kops.CloudProviderGCE {
+	// It will land on nodes labeled cloud.google.com/metadata-proxy-ready=true when the feature is enabled
+	if b.Cluster.Spec.CloudProvider.GCE != nil && b.Cluster.Spec.CloudProvider.GCE.MetadataProxy != nil && fi.ValueOf(b.Cluster.Spec.CloudProvider.GCE.MetadataProxy.Enabled) {
```

Comment:
MetadataProxy is guaranteed to be non-`nil` at this point. But if it were `nil` then the proxy would be enabled.

Same for MetadataProxy.Enabled.